src/System.Management.Automation/engine/parser/TypeResolver.cs

```diff
@@ -117,8 +117,10 @@ public AmbiguousTypeException(TypeName typeName, IEnumerable<string> candidates)
                             }
                         }
                     }
-                    else
+                    else if (!assembly.IsDynamic)
```

Comment:
I'm not convinced this is the only case - what about unsigned assemblies - no key, same version?